File: 1-Two-Sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        //hash map storing nums as keys and the indices of nums as values
        let mut hm: std::collections::HashMap<i32, i32> = std::collections::HashMap::new();
        
        //for each in input array
        for i in 0..nums.len() {
            //if the hashmap contains the number needed to create target
            if hm.contains_key(&(target - nums[i])) {
                //return the vector with indices need to create target
                return vec![hm[&(target-nums[i])], i as i32];
            }
            //insert the current number and index into hashmap
            hm.insert(nums[i], i as i32);
        }
        return vec![];
    }
}

File: 1-Two-Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        if(nums == null || nums.Length < 2 )
            return new int[2];

        Dictionary<int,int> check = new Dictionary<int,int>() ;

        for(int i = 0; i < nums.Length; i++)
        {
            if(check.ContainsKey(target - nums[i]))
            {
                return new int[]{check[target-nums[i]], i};
            }
            check[nums[i]] = i;
        }
        return new int[]{0,0};
    }
}